### DISCUSSION TITLE
Locked Features and Monetization Models

### DISCUSSION THREAD

**Discussion by x80486** [ariga/atlas#2871] [General]
I've been testing Atlas for a few days now. In my opinion, it's at the top in its category in the Go world. I've tried many solutions in this area and the only one that checks all the boxes is Atlas. It's the closest I was able to find (in **Versioned** mode) to Liquibase or Flyway for enterprise-grade solutions, where simplicity, correctness, and performance are crucial factors.

The organization I currently work for, like many others, operates in a (hysteric) secure environment where external Internet access is restricted. This constraint takes another level in the Continuous Integration and Delivery (CI/CD) platform, which cannot connect to online services.

As a result (you may have guessed :sweat_smile:), due to the `login` constrain, there are several commonly used features that can't be used, so in my case, it's going to be hard to propose Atlas for a new effort I'm leading to consider Go for new applications. Moreover, this may well be the status for others — willing to pay for premium features, but unable to use the solution because of locking these features behind an online service due to security policies and/or infrastructure limitations. Basically, uploading _sensitive data_ to the cloud (even for migration management) raises security concerns, and many organizations won't consider certain solutions because their top priority is keeping their data secure.

On the other hand, I understand you need to find ways to capitalize on your work — which is really awesome to this date.

So I just wanted to share this and perhaps get your input in this matter.

**Comment by rotemtam** [comment]
Hi

Thanks for reaching out. We have solutions for customers who need on premises/air-gapped deployments.

Feel free to ping me on our discord server or via hello@ariga.io and we will happily help you with whatever you need 

Rotem
